src/api/feedDiscovery/src/lib/discover.js

```diff
@@ -0,0 +1,74 @@
+const got = require('got');
+const cheerio = require('cheerio'); // jQuery for node
+
+// Helper function to check the validity of the blog url
+function isValidUrl(blogUrl) {
+  // Reference source for regEx string
+  // https://gist.github.com/dperini/729294
+  const validUrlRegex = /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z0-9\u00a1-\uffff][a-z0-9\u00a1-\uffff_-]{0,62})?[a-z0-9\u00a1-\uffff]\.)+(?:[a-z\u00a1-\uffff]{2,}\.?))(?::\d{2,5})?(?:[/?#]\S*)?$/i;
+  return validUrlRegex.test(blogUrl);
+}
+
+// Helper function to return the feed url of a given blog url
+async function getFeedUrl(blogUrl) {
+  try {
+    const res = await got.get(blogUrl);
+    // Load the html document
+    const $ = cheerio.load(res.body);
+    // Extract the list of rss link elements
+    const rss = $('link[type="application/rss+xml"]').clone();
+    // Extract the list of atom link elements
+    const atom =
+      $('link[type="application/atom+xml"]').clone() ||
+      $('link[type="application/x.atom+xml"]').clone() ||
+      $('link[type="application/x-atom+xml"]').clone();
+    const feedList = rss.length ? rss : atom; // If rss has any content (i.e. it actually exists), assign its value to elem, otherwise assign the value of atom
+    const feedUrl = feedList[0].attribs.href;
+    return feedUrl;
+  } catch (e) {
+    throw new Error();
+  }
+}
+
+// A middleware to check if Url provided is valid
+function checkValidUrl(req, res, next) {
+  // If the URL is invalid, return 400 error
+  if (!isValidUrl(req.body.blogUrl)) {
+    res.status(400).send('Invalid Blog URL Provided');
+  }
+  // Else, continue
+  next();
+}
+
+// A middleware to check if the URL is a valid web page, 200 status with text/html type
+async function checkValidWebPage(req, res, next) {
+  try {
+    const { statusCode, headers } = await got.head(req.body.blogUrl);
+    const contentType = headers['content-type'];
+    // If status code is not 200 or content-type is not text/html then send 400 error
+    if (statusCode !== 200 || !contentType.includes('text/html')) {
+      res.status(400).send();
```

Comment:
Create a proper HTTP error, like this: https://github.com/Seneca-CDOT/satellite/blob/master/index.js#L113